p2p/pre2p/raintree/network.go

```diff
@@ -33,6 +37,9 @@ type rainTreeNetwork struct {
 
 	// TECHDEBT(drewsky): What should we use for de-duping messages within P2P?
 	mempool types.Mempool
+
+	// TODO(team): think about proper way to keep track of p2p seen msgs hashes
+	msgspool map[string]bool
```

Comment:
Have you looked at how `mempool` defined above is used?

Curious to understand why it can't be reused / is not enough.